### ISSUE TITLE
Numerous problems with subreddit selection, sort order, disappearing tasks, and date format

### ISSUE THREAD

**Issue by dziban303** [VictorColomb/warplaneporn-lockscreen#5]
I'm having several issues with it. Firstly, Get-date wants the full option `-Format` instead of the `-f` shortcut in Powershell v7.1 preview; I created a pull request for that ( #4 ). 

```pwsh
....
Get-Date: C:\Users\dziba\Documents\PowerShell\Modules\WarplanepornLockscreen\1.0.0\WarplanepornLockscreen.psm1:467
Line |
 467 |          $date = Get-date -f "dd/MM/yyyy HH:mm:ss"
     |                           ~~
     | Parameter cannot be processed because the parameter name 'f' is ambiguous. Possible matches include:
     | -Format -FromUnixTime.

WarplanepornLockscreen is installed
.....
```

Now, in both Powershell v7.0.1 and v7.1.0-preview.3, regardless of what subreddit I specify, it only checks WarplanePorn—even if I don't include it:

```pwsh
PS C:\Windows\System32> WarplanepornLockscreen -config
Do you want to include nsfw posts in the potential wallpapers ? (y/n) (default no): y
How to sort posts ? (top|hot|new) (leave empty for hot): top
What subreddits would you like to see wallpapers from ?
Enter one without the r/ prefix, press enter then the next one. Leave a line empty to confirm.
Just press ENTER for default (warplaneporn)
Subreddit 1: warshipporn
Checking subreddit warshipporn...
Subreddit 2: mapporn
Checking subreddit mapporn...
Subreddit 3: atomicporn
Checking subreddit atomicporn...
Subreddit 4:
List of configured subreddits : warplaneporn
Will choose image from subreddit warplaneporn, sorting by hot
Downloading image at https://i.redd.it/poz6wyh28a651.jpg
Setting lockscreen background
PS C:\Windows\System32>
```

Contents of `config.json` after running the above:

```json
{
  "nsfw": true,
  "subreddits": [
    "warshipporn",
    "mapporn",
    "atomicporn"
  ],
  "sort": "hot"
}
```

It still only downloads the same image from WPP. I earlier tried going with the default `warplaneporn` sub, but changed the sort option to `new` or `top`; it doesn't accept these options either, it's stuck on `hot`.

Then it sometimes forgets that the daily task is installed. Immediately after running the above config task, I tried again:

```pwsh
PS C:\Windows\System32> WarplanepornLockscreen -config
The module was never installed, would you like to install it ? (y/n) : y

The task that is about to be created needs an administrator password to work
Enter administrator username :
....
```
The task is, in fact, showing up in Task Scheduler. Presumably the `"installed": true,` string in `config.json` is getting deleted erroneously during some, but not all, `-config` tasks.

So then I tried to adjust the subreddits with `-add` and `-remove`:

```pwsh
PS C:\Windows\System32> WarplanepornLockscreen -list
warplaneporn
PS C:\Windows\System32> WarplanepornLockscreen -add warshipporn
Subreddit warshipporn was added to configuration
PS C:\Windows\System32> WarplanepornLockscreen -list
warplaneporn warshipporn
PS C:\Windows\System32> WarplanepornLockscreen -remove warplaneporn
Subreddit warplaneporn was removed from configuration
PS C:\Windows\System32> WarplanepornLockscreen -list
warshipporn
PS C:\Windows\System32> WarplanepornLockscreen
List of configured subreddits : warplaneporn
Will choose image from subreddit warplaneporn, sorting by hot
Downloading image at https://i.redd.it/poz6wyh28a651.jpg
Setting lockscreen background
PS C:\Windows\System32>
```

And it still goes for WarplanePorn. So I want to start over by uninstalling:

```pwsh
PS C:\Windows\System32> WarplanepornLockscreen -uninstall
Removed registry key
Get-ScheduledTask: C:\Users\dziba\Documents\PowerShell\Modules\WarplanepornLockscreen\1.0.0\WarplanepornLockscreen.psm1:407
Line |
 407 |          if (Get-ScheduledTask -TaskName "WarplanepornLockscreen") {
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'WarplanepornLockscreen'.
     | Verify the value of the property and retry.

Unregistered task
Uninstalled module
```

I didn't check to see if the task was still there immediately before uninstalling, but it had been there earlier. The WarplanepornLockscreen folder is still present in Task Scheduler.

![image](https://user-images.githubusercontent.com/7931765/85245026-fc5fd400-b40b-11ea-8ec1-3f948195d5ad.png)

I'll try installing it again later, and will update with results.

**Comment by dziban303** [comment]
OK, after reinstalling, the issues persist. I didn't select warplaneporn at all, yet that's what I get.

```pwsh
PS C:\Windows\System32> Get-ExecutionPolicy
Unrestricted
PS C:\Windows\System32> Get-ExecutionPolicy -Scope Process
Unrestricted
PS C:\Windows\System32> Install-Module -Name WarplanepornLockscreen

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): y
PS C:\Windows\System32> WarplanepornLockscreen -install

The task that is about to be created needs an administrator password to work
Enter administrator username : dz@nunya.biz
Enter password for that user : *****************************
Username and password are correct
Unregistering any existing scheduled task
Registering new task
WarplanepornLockscreen is installed

Do you want to include nsfw posts in the potential wallpapers ? (y/n) (default no): y
How to sort posts ? (top|hot|new) (leave empty for hot): top
What subreddits would you like to see wallpapers from ?
Enter one without the r/ prefix, press enter then the next one. Leave a line empty to confirm.
Just press ENTER for default (warplaneporn)
Subreddit 1: warshipporn
Checking subreddit warshipporn...
Subreddit 2: atomicporn
Checking subreddit atomicporn...
Subreddit 3: mapporn
Checking subreddit mapporn...
Subreddit 4: 

List of configured subreddits : warplaneporn
Will choose image from subreddit warplaneporn, sorting by top
Downloading image at https://i.redd.it/poz6wyh28a651.jpg
Setting lockscreen background
PS C:\Windows\System32>

**Comment by dziban303** [comment]
I just tried installing the V1 executable archive, and it did the same thing with selecting only warplaneporn whether it's specified or not. I don't know what else to do.

**Comment by VictorColomb** [comment]
It's the same module code in all the installers so if one doesn't work none of them will.

I never tested on PS7 so it's possible that there are compatibility problems. I'll look at your PR, thanks.

There should be an `"installed" = true` in the config, this is weird. That's why you're getting the prompt to install.  
There's clearly a problem with the config file. I remember that it worked on PS 5.  
I'll install PS 7 and try to reproduce.

PS: if there's a task folder and that the installer didn't fail, the task was most likely created successfully

**Comment by dziban303** [comment]
The .exe installer actually put it in Powershell 5, and the problems were occurring there as well—so PS version doesn't seem to matter (aside from the Get-date switch verbiage). 

Running the program in one-off mode while explicitly specifying subreddits, e.g., 

```pwsh
PS C:\Users\dziba> WarplanepornLockscreen -subreddits WarshipPorn, WarplanePorn, MapPorn -nsfw -sort top
```

does seem to work, so I've replaced the taskschd action's switches with a similar string.

**Comment by VictorColomb** [comment]
Thanks for the PR ! Here's what I've got so far :

* The `"installed": true` was getting removed sometimes, my bad
* I think I lost a commit along the way so the config file was not properly read, thus r/warplaneporn staying default
    Fixed in [`v1.0.1`](https://github.com/viccol961/warplaneporn-lockscreen/releases/tag/v1.0.1)
* However, the sort and nsfw config entries work properly on my side (both PS 5 and 7). I couldn't reproduce...
    It seems to have worked for you [here](#issuecomment-647248636) though.
    Maybe try again with the patch and tell me if it still doesn't work.

Edit: see [`v1.0.3`](https://github.com/viccol961/warplaneporn-lockscreen/releases/tag/v1.0.3) that I just released instead.

I'll let you close the issue whenever you think everything is fixed.

**Comment by dziban303** [comment]
Everything previously mentioned has been addressed in v1.0.3.

However, when I reinstalled v1.0.3 on PS v5, I got an error upon running `-config` which said the log at `C:\Users\dziba\Documents\WindowsPowerShell\Modules\WarplanepornLockscreen\log.txt` wasn't accessible; it seems the `\WarplanepornLockscreen` directory hadn't been created. Rather than troubleshoot it—I had a long day—I just removed WarplanepornLockscreen (I'm gonna start calling it WPP-LS from now on, it's too long a name!) and then installed it on PSv7. I ran it through its paces and it worked like a champ.

Thanks!
